src/modm/platform/can/sam-mcan/message_ram.hpp

```diff
@@ -0,0 +1,337 @@
+/*
+ * Copyright (c) 2019, 2023, Raphael Lehmann
+ * Copyright (c) 2021, Christopher Durand
+ * Copyright (c) 2022, Rasmus Kleist Hørlyck Sørensen
+ *
+ * This file is part of the modm project.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+// ----------------------------------------------------------------------------
+
+#pragma once
+
+#include <cstdint>
+#include <concepts>
+#include <span>
+
+#include <modm/math/utils/bit_constants.hpp>
+#include <modm/architecture/interface/register.hpp>
+
+/// @cond
+
+namespace modm::platform::fdcan
+{
+
+
+struct MessageRamConfig
+{
+	uint32_t filterCountStandard;
+	uint32_t filterCountExtended;
+	uint32_t rxFifo0Elements;
+	uint32_t rxFifo0ElementSize;
+	uint32_t rxFifo1Elements;
+	uint32_t rxFifo1ElementSize;
+	uint32_t rxBufferElements;
+	uint32_t rxBufferElementSize;
+	uint32_t txEventFifoEntries;
+	uint32_t txFifoElements;
+	uint32_t txFifoElementSize;
+};
+
+constexpr MessageRamConfig defaultMessageRamConfig
+{
+	.filterCountStandard 	= 1,
+	.filterCountExtended	= 1,
```

Comment:
Default is one filter? That is not a lot.

Reply:
How many filters would you suggest?
The examples in modm use at most 1 standard and one extended filter...

Reply:
My actual SAMV70 production code receives 11 different ids which can be expressed with 4 filters in that particular case. Those SAM Cortex-M7 devices all have plenty of RAM. Having 4-8 filters by default certainly wouldn't hurt much. Trying to use more filters than available silently fails at runtime. That can take a long time to debug if you haven't read the docs carefully.

Reply:
The `setFilter[…]()` functions return false if the filter index is out of range, but the user code needs to check for the return value...

I'll increase the number of filters to 8 each.